src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameInfo.cs

```diff
@@ -23,14 +23,17 @@ namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers.DeclarationName
         private static readonly ImmutableArray<SymbolKindOrTypeKind> s_propertySyntaxKind =
             ImmutableArray.Create(new SymbolKindOrTypeKind(SymbolKind.Property));
 
-        public readonly ImmutableArray<SymbolKindOrTypeKind> PossibleSymbolKinds;
+        private readonly ImmutableArray<SymbolKindOrTypeKind> _possibleSymbolKinds;
+
         public readonly DeclarationModifiers Modifiers;
         public readonly Accessibility? DeclaredAccessibility;
 
         public readonly ITypeSymbol? Type;
         public readonly IAliasSymbol? Alias;
         public readonly ISymbol? Symbol;
 
+        public ImmutableArray<SymbolKindOrTypeKind> PossibleSymbolKinds => _possibleSymbolKinds.NullToEmpty();
```

Comment:
Is there any reason to do this as a getter instead of just doing this in the constructor and assigning once?

Reply:
this is a struct, so the constructor is not always called.  this makes us follow the pattern used in things like SymbolInfo where the default-struct is the proper representation of 'no value'.